src/app/campaign/flight/flight-targets-form.component.ts

```diff
@@ -73,6 +73,8 @@ export class FlightTargetsFormComponent implements ControlValueAccessor {
   writeValue(targets: FlightTarget[]) {
     this.setCodeOptions(targets);
     this.targets = new FormArray((targets || []).map(this.newFlightTarget));
+    // TODO: wouldn't this be leaking subscriptions? Instead of new'ing the FormArray and subscribing on each update,
+    //  it should update/add/remove the controls in the array
```

Comment:
This was another thing that concerned me with how we were subscribing each time the form was updated.